src/Helper.js

```diff
@@ -355,7 +356,15 @@ class Helper {
 
     // Get states whose entity-id starts with the given string.
     const stateEntities = Object.values(this.#hassStates).filter(
-        state => state.entity_id.startsWith(`${domain}.`),
+        state => {
+          if (deviceClass
```

Comment:
Can the return conditions rewritten to a ternary operator statement?
E.g.: `return deviceClass && state.attributes.device_class === deviceClass ? state.entity_id.startsWith(`${domain}.`) : false;`